core/src/test/java/org/apache/calcite/sql/type/RelDataTypeSystemTest.java

```diff
@@ -118,6 +126,13 @@ private static final class CustomTypeSystem extends RelDataTypeSystemImpl {
     @Override public int getMaxNumericPrecision() {
       return 38;
     }
+
+    @Override public int getMaxPrecision(SqlTypeName typeName) {
+      if (typeName == SqlTypeName.TIMESTAMP) {
```

Comment:
why not just have a `MAX_TIMESTAMP_PRECISION`? Am I wrong in saying that everytime we look for the max timestamp precision we're just retrieving the max datetime and then adding 3? In that case we should just define it as a constant and then this becomes unnecessary.

Reply:
I named this constant CUSTOM_MAX_TIMESTAMP_PRECISION just to be explicit that it's testing a custom max.